ftag/flavour.py

```diff
@@ -43,6 +43,17 @@ def __str__(self) -> str:
 class FlavourContainer:
     flavours: dict[str, Flavour]
 
+    def __init__(self, yaml_path: Path = None):
```

Comment:
If you are going to define `__init__` this probably shouldn't be a dataclass.

I think it's nice to keep have the option for people to define directly from a python dict, so I'd suggest to keep this as a dataclass but add `from_yaml` class method as done [here](https://github.com/umami-hep/umami-preprocessing/blob/d0b8a25f029a461499c782a5508be53bb4562e63/upp/classes/components.py#L104).

Reply:
totally, I forget about all these magic decorator features 😅